static/app/views/eventsV2/savedQuery/index.tsx

```diff
@@ -12,22 +15,74 @@ import Banner from 'sentry/components/banner';
 import Button from 'sentry/components/button';
 import ButtonBar from 'sentry/components/buttonBar';
 import {CreateAlertFromViewButton} from 'sentry/components/createAlertButton';
-import DropdownControl from 'sentry/components/dropdownControl';
 import InputControl from 'sentry/components/forms/controls/input';
 import {Hovercard} from 'sentry/components/hovercard';
+import {Overlay, PositionWrapper} from 'sentry/components/overlay';
 import {IconDelete, IconStar} from 'sentry/icons';
 import {t} from 'sentry/locale';
 import space from 'sentry/styles/space';
 import {Organization, Project, SavedQuery} from 'sentry/types';
 import {trackAnalyticsEvent} from 'sentry/utils/analytics';
 import EventView from 'sentry/utils/discover/eventView';
 import {getDiscoverLandingUrl} from 'sentry/utils/discover/urls';
+import useOverlay from 'sentry/utils/useOverlay';
 import withApi from 'sentry/utils/withApi';
 import withProjects from 'sentry/utils/withProjects';
 import {handleAddQueryToDashboard} from 'sentry/views/eventsV2/utils';
 
 import {handleCreateQuery, handleDeleteQuery, handleUpdateQuery} from './utils';
 
+type SaveAsDropdownProps = {
+  disabled: boolean;
+  modifiedHandleCreateQuery: (e: React.MouseEvent<Element>) => void;
+  onChangeInput: (e: React.FormEvent<HTMLInputElement>) => void;
+  queryName: string;
+};
+
+function SaveAsDropdown({
+  queryName,
+  disabled,
+  onChangeInput,
+  modifiedHandleCreateQuery,
+}: SaveAsDropdownProps) {
+  const {isOpen, triggerProps, overlayProps, arrowProps} = useOverlay({});
+  const theme = useTheme();
+
+  return (
+    <div>
+      <Button icon={<IconStar />} aria-disabled={disabled} {...triggerProps}>
+        Save as
+      </Button>
+      <AnimatePresence>
+        {isOpen && (
+          <FocusScope contain restoreFocus autoFocus>
+            <PositionWrapper zIndex={theme.zIndex.dropdown} {...overlayProps}>
+              <StyledOverlay arrowProps={arrowProps} animated>
+                <ButtonSaveInput
+                  type="text"
+                  name="query_name"
+                  placeholder={t('Display name')}
+                  value={queryName || ''}
+                  onChange={onChangeInput}
+                  disabled={disabled}
+                />
+                <Button
+                  onClick={modifiedHandleCreateQuery}
+                  priority="primary"
+                  disabled={disabled || !queryName}
+                  style={{width: '100%'}}
```

Comment:
should we use emotion here to style the button?

Reply:
True, updated in https://github.com/getsentry/sentry/pull/36402/commits/81bdc9ee8fb10a631a4e1166d7e76a94dd6bfd20